---
title: "First part: Importing Data & Pseudo-alignment"
author: "Hannia Larino"
date: "2025-04-21"
output: html_document
editor_options: 
  markdown: 
    wrap: 72
---

Location of folder:
/home/shared/8TB_HDD_02/hannia/SeaCucumber/PRJNA848687_fastq Fastq
files: raw data generated from RNA-seq experiments are typically stored
in fastq files, which contain millions of short sequencing reads that
need to be preprocessed before performing differential gene expression
analysis. Using Kallisto to analyze RNA seq data and ID DEG.

Components of fastq files: A FASTQ file has four line-separated fields
per sequence: *Field 1 begins with a '\@' character and is followed by a
sequence identifier and an optional description (like a FASTA title
line).* Field 2 is the raw sequence letters. *Field 3 begins with a '+'
character and is optionally followed by the same sequence identifier
(and any description) again.* Field 4 encodes the quality values for the
sequence in Field 2, and must contain the same number of symbols as
letters in the sequence. --------- Step 1. Downlaod FastQC software from
<https://www.bioinformatics.babraham.ac.uk/projects/download.html#fastqc>
#downloaded and imported manually using "upload". Next time do it on the
command line.

Professor downloaded this data into my directory:
<https://www.ncbi.nlm.nih.gov/bioproject/PRJNA848687/>

About the sample ID of .fastq files:
<https://www.ncbi.nlm.nih.gov/biosample?LinkName=bioproject_biosample_all&from_uid=848687>
Paired-end files labeled as _1 and _2 with ID code: 
1.  SRR19635628 = Control 1 

2.  SRR19635629 = Control 2

3.  SRR19635630 = Control 3

4.  SRR19635631 = 26 deg Cels

5.  SRR19635632 = 26 deg Cels

6.  SRR19635633 = 26 deg Cels

7.  SRR19635634 = 26 deg Cels

8.  SRR19635635 = 26 deg Cels

9.  SRR19635636 = 26 deg Cels

10. SRR19635637 = 30

11. SRR19635638 = 30

12. SRR19635639 = 30

**RUN FastQC** Running FastQC program on PRJNA848687_fastq folder. This
will generate a HTML report that can be viewed in a web browser. There
are 12 unique samples total. Line 28- this is how you run the fastqc
program, similar to blast.

```{bash}

/home/shared/8TB_HDD_02/hannia/SeaCucumber/FastQC/fastqc \
/home/shared/8TB_HDD_02/hannia/SeaCucumber/PRJNA848687_fastq/*.fastq \
-o /home/shared/8TB_HDD_02/hannia/SeaCucumber/output
```

#The QC results show that all sampels have a red x for per base sequence
content and sequence duplication levels.

#2. Get a reference genome of sea cucumber (Apostichopus japonicus)
from: <https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_037975245.1/>

##curl into the SeaCucumber folder

```{bash}
URL="https://api.ncbi.nlm.nih.gov/datasets/v2/genome/accession/GCF_037975245.1/download?include_annotation_type=GENOME_FASTA&include_annotation_type=GENOME_GFF&include_annotation_type=RNA_FASTA&include_annotation_type=CDS_FASTA&include_annotation_type=PROT_FASTA&include_annotation_type=SEQUENCE_REPORT&hydrated=FULLY_HYDRATED"

curl -L "$URL" -o /home/shared/8TB_HDD_02/hannia/SeaCucumber/GCF_037975245.1_ref.zip
```

###unizp GCF_037975245.1_ref

```{bash}
cd /home/shared/8TB_HDD_02/hannia/SeaCucumber
unzip GCF_037975245.1_ref.zip -d GCF_037975245.1_ref 
```

####explore contents

```{bash}
cd /home/shared/8TB_HDD_02/hannia/SeaCucumber/GCF_037975245.1_ref/ncbi_dataset/data/GCF_037975245.1
ls

```

**2. (Pseudo - alignment)** checking if single end or paired

```{bash}
head -4 /home/shared/8TB_HDD_02/hannia/SeaCucumber/PRJNA848687_fastq/*.fastq          

```

#making Kallisto index first b4 pseudo alignment (needed for Kallisto
quant) #making index from ref transcriptome (rna.fna file) #answers
above show that these files are paired-end FASTQ files #index file:
compressed data structure that allows fast pseudoalignment of RNA-Seq
reads to a reference transcriptome.This is for faster alignment, does
not align entire genome but instead matching transcripts.

```{bash}
/home/shared/kallisto/kallisto index \
-i /home/shared/8TB_HDD_02/hannia/SeaCucumber/index.idx \
/home/shared/8TB_HDD_02/hannia/SeaCucumber/GCF_037975245.1_ref/ncbi_dataset/data/GCF_037975245.1/rna.fna
```



**3. Differential Expression Analysis (Kallisto)**

*Already created index, next step making output folder for kallisto
data.*

```{r, engine='bash'}
cd /home/shared/8TB_HDD_02/hannia/SeaCucumber 
pwd
mkdir /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01 
#mkdir makes new folder called kallisto01 up one directory level 
```


*Running kallisto quantification command*: tells you how much of each transcript
(mRNA isoform) is present in your RNA-Seq sample, based on the
sequencing reads you provide. Quantifying abundance of expressed genes
from samples.

```{r, engine='bash'}
find /home/shared/8TB_HDD_02/hannia/SeaCucumber/PRJNA848687_fastq/*_1.fastq \
| xargs -n1 basename \
| sed 's/_1\.fastq$//' \
| xargs -I{} /home/shared/kallisto/kallisto quant \
-i /home/shared/8TB_HDD_02/hannia/SeaCucumber/index.idx \
-o /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/{} \
-t 40 \
--paired \
/home/shared/8TB_HDD_02/hannia/SeaCucumber/PRJNA848687_fastq/{}_1.fastq \
/home/shared/8TB_HDD_02/hannia/SeaCucumber/PRJNA848687_fastq/{}_2.fastq
```


*Using abundance estimates from Kallisto quant to create gene expression
matrix*

```{r, engine='bash'}
perl /home/shared/trinityrnaseq-v2.12.0/util/abundance_estimates_to_matrix.pl \
  --est_method kallisto \
  --gene_trans_map none \
  --out_prefix /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01 \
  --name_sample_by_basedir \
  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635628/abundance.tsv \
  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635629/abundance.tsv \
  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635630/abundance.tsv \
  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635631/abundance.tsv \
  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635632/abundance.tsv \
  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635633/abundance.tsv \
  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635634/abundance.tsv \
  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635635/abundance.tsv \
  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635636/abundance.tsv \
  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635637/abundance.tsv \
  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635638/abundance.tsv \
 /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635639/abundance.tsv
```

#chunk above gives error but created .matrix anyway
