What is your prior experience in this discipline?
I have had experience using BLAST and R studio in several biology and math classes. However, this is my first time using BLAST through R. It is also my first time using Jupyter and Github. 

What do you hope to get out of this class?
I hope to learn how to use R studio to do a mass analyses of genetic data using BLAST. For example, if I were to collect DNA of microbial species to to a microbiome analysis of an invertebrate, I would want to know how to use R to complete that analysis in the most efficient way possible. I also hope to feel more comfortable with the different programs we are using to feel I can replicate it on my own. 

This class is strongly rooted in an independent project related to genomic analyses. What specific project do you have in mind? If you do not have any data or preference, data can be provided / aquired. If you do not have a specfic project, what approach would you like to master as part of this class?

I would like to collect genetic data that demonstrates how exposure to pollution impacts the gene expression of invertebrates. Perhaps genes that are important to embryonic development or cell metabolism. The species is not super important to me but I would like to see if there is genetic data on barnacles because they tend to be resilient to harsh environments. 

What are two things you found most useful from the reading?
The "Navigating Files and Directories" reading was useful to me, particularly to understand the directory terminology. Such as relative path (cd ../../), absolute path (/home/../..), and home directory. 

I also found the functions of the basic commands on both readings, such as cd or pwd, to be really helpful. I've only used R languge so I am trying to catch up to learn the basic commands of {bash}. 
