{
  "cells": [
    {
      "cell_type": "raw",
      "metadata": {},
      "source": [
        "---\n",
        "title: \"Compendium of FISH546 project\"\n",
        "author: Hannia Larino\n",
        "date: \"Spring 2025\"\n",
        "format: revealjs\n",
        "editor: visual\n",
        "execute:\n",
        "  kernel: bash\n",
        "  eval: false   \n",
        "---"
      ],
      "id": "9d77dc03"
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## FISH546 Differential Gene Expression Analysis Project\n",
        "\n",
        "## Purpose\n",
        "\n",
        "I am using RNA-seq data taken from Sea Cucumbers (*Apostichopus japonicus*) that were treated under 2 different temperatures (26°C & 30°C). The purpose being to conduct DGE analysis to determine the biological responses that heat stress induces on this organism. Data was obtained from the NIH website, done by researchers in Xu et al., at Qingdao Agricultural University (2023).\n",
        "\n",
        "## Background\n",
        "\n",
        "As climate change develops, water temperature rising becomes an increasing threat to marine life (Lindsey & Dahlman, 2023). Researchers have found 30°C to be a lethal temperature for the marine invertebrate *Apostichopus japonicus* (SOURCE). In the research study by Xu et al., 2023, they cite aquaculture of sea cucumbers as a motive to test the effect that heat stress has on gene expression. The focus of Xu et al., is to analyze the effects that heat stress has on heat-respondent proteins.\n",
        "\n",
        "## Methods\n",
        "\n",
        "The design experiment consisted of 3 controls maintained at 18°C. Six sub-lethal temperature treatment groups (26°C) and three lethal temperature treatment groups (30°C). The treatment groups went through a temperature-rise process from 18°C to 26°C or to 30°C respectively, with a rate of 2°C per hour by using a heating rod. The 26°C treatment group was able to be kept at that temperature for 6 hours and 48 hours, creating two treatment groups within the 26°C treatment group (6 hrs vs 48 hrs). The 30°C treatment group was only kept at 30°C for 6 hours. The researchers site full mortality when they did try to create a 48 hr group within this temperature treatment. Intestine tissue was obtained for RNA-Seq analysis.\n",
        "\n",
        "## Code\n",
        "\n",
        "Broken in 4 parts.\n",
        "\n",
        "## Part 1: Obtain data & conduct quality check (QC)\n",
        "\n",
        "FastQ files were obtained from NCBI. Accession code being: PRJNA848687. More information about the data can be found [here](https://www.ncbi.nlm.nih.gov/bioproject/848687). Go [here](https://www.ncbi.nlm.nih.gov/biosample?LinkName=bioproject_biosample_all&from_uid=848687) if you want to access information about the individual files (12 total).\n",
        "\n",
        "The quality check was done using FastQC:\n",
        "\n",
        "\n",
        "```{bash}\n",
        "#| echo: true\n",
        "/home/shared/8TB_HDD_02/hannia/SeaCucumber/FastQC/fastqc \\\n",
        "/home/shared/8TB_HDD_02/hannia/SeaCucumber/PRJNA848687_fastq/*.fastq \\\n",
        "-o /home/shared/8TB_HDD_02/hannia/SeaCucumber/output\n",
        "```\n",
        "\n",
        "\n",
        "Results: HTML results of QC did not show presence of outliers that needed to be removed. Refer to \\~SeaCucumber/output_fastqc for the html files.\n",
        "\n",
        "## Part 2: Pseudo-alignment\n",
        "\n",
        "The reference genome of *Apostichopus japonicus* for the pseudo-alignment was obtained from NCBI and can be accessed [here](https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_037975245.1/). The **NCBI RefSeq assembly ID is:** GCF_037975245.1.\n",
        "\n",
        "Pseudo-aligment was done using Kallisto. An index was created first (1). Then the pseudo-alignment was completed using Kallisto quant for paired-end reads (2).\n",
        "\n",
        "1.  \n",
        "\n",
        "\n",
        "```{bash}\n",
        "#| echo: true\n",
        "/home/shared/kallisto/kallisto index \\\n",
        "-i /home/shared/8TB_HDD_02/hannia/SeaCucumber/index.idx \\\n",
        "/home/shared/8TB_HDD_02/hannia/SeaCucumber/GCF_037975245.1_ref/ncbi_dataset/data/GCF_037975245.1/rna.fna\n",
        "```\n",
        "\n",
        "\n",
        "2 .\n",
        "\n",
        "\n",
        "```{bash}\n",
        "#| echo: true\n",
        "find /home/shared/8TB_HDD_02/hannia/SeaCucumber/PRJNA848687_fastq/*_1.fastq \\\n",
        "| xargs -n1 basename \\\n",
        "| sed 's/_1\\.fastq$//' \\\n",
        "| xargs -I{} /home/shared/kallisto/kallisto quant \\\n",
        "-i /home/shared/8TB_HDD_02/hannia/SeaCucumber/index.idx \\\n",
        "-o /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/{} \\\n",
        "-t 40 \\\n",
        "--paired \\\n",
        "/home/shared/8TB_HDD_02/hannia/SeaCucumber/PRJNA848687_fastq/{}_1.fastq \\\n",
        "/home/shared/8TB_HDD_02/hannia/SeaCucumber/PRJNA848687_fastq/{}_2.fastq\n",
        "```\n",
        "\n",
        "\n",
        "## Part 3: Creating a gene expresison matrix \n",
        "\n",
        "\n",
        "```{bash}\n",
        "#| echo: true\n",
        "perl /home/shared/trinityrnaseq-v2.12.0/util/abundance_estimates_to_matrix.pl \\\n",
        "  --est_method kallisto \\\n",
        "  --gene_trans_map none \\\n",
        "  --out_prefix /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01 \\\n",
        "  --name_sample_by_basedir \\\n",
        "  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635628/abundance.tsv \\\n",
        "  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635629/abundance.tsv \\\n",
        "  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635630/abundance.tsv \\\n",
        "  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635631/abundance.tsv \\\n",
        "  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635632/abundance.tsv \\\n",
        "  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635633/abundance.tsv \\\n",
        "  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635634/abundance.tsv \\\n",
        "  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635635/abundance.tsv \\\n",
        "  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635636/abundance.tsv \\\n",
        "  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635637/abundance.tsv \\\n",
        "  /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635638/abundance.tsv \\\n",
        " /home/shared/8TB_HDD_02/hannia/SeaCucumber/output/kallisto_01/SRR19635639/abundance.tsv\n",
        "```\n",
        "\n",
        "\n",
        "## Part 4: DESeq2 for DGE analysis \n",
        "\n",
        "\n",
        "```{bash}\n",
        "#| echo: true\n",
        "\n",
        "\n",
        "```\n",
        "\n",
        "\n",
        "## Results so far (week 8)\n",
        "\n",
        "## Plan for next 2 weeks\n",
        "\n",
        "## References\n",
        "\n",
        "Dahlman, L., & Lindsey, R. (2023). Climate change: Ocean heat content. NOAA Climate.gov. https://www.climate.gov/news-features/understanding-climate/climate-change-ocean-heat-content."
      ],
      "id": "e65f23be"
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}