---
title: "07-CG motifs"
format: html
editor: visual
---

```{r}
install.packages("seqinr")
library(seqinr)

setwd("/home/shared/8TB_HDD_02/hannia/SeaCucumber/GCF_037975245.1_ref/ncbi_dataset/data/GCF_037975245.1")

# Replace 'input.fasta' with the name of your multi-sequence fasta file

input_file <- "rna.fna"
sequences <- read.fasta(input_file)

```

You can add options to executable code like this

```{r}
input_file
sequences
```

```{r}
# Set the seed for reproducibility (optional)
set.seed(42)

number_of_sequences_to_select <- 10

if (length(sequences) < number_of_sequences_to_select) {
  warning("There are fewer than 10 sequences in the fasta file. All sequences will be selected.")
  number_of_sequences_to_select <- length(sequences)
}

selected_indices <- sample(length(sequences), number_of_sequences_to_select)
selected_sequences <- sequences[selected_indices]
selected_sequences

```

```{r}
# Replace 'output.fasta' with your desired output file name
output_file <- "output/outputCGmotifs.fasta"
write.fasta(selected_sequences, names(selected_sequences), output_file, open = "w")
``
```

```{bash}
#likely will not need; fix issue where gff and fa name did not match
# sed -i 's/>lcl|/>/g' ../output/10_seqs.fa
```

```{bash}
#needed downstream for IGV
/home/shared/samtools-1.12/samtools faidx \
output/outputCGmotifs.fasta #result gives .fai file in output folder 
```

```{bash}
fuzznuc -sequence output/outputCGmotifs.fasta -pattern CG -rformat gff -outfile output/CGoutput.gff
```
